homeassistant/components/template/trigger_entity.py

```diff
@@ -57,3 +58,21 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         self._process_data()
         self.async_write_ha_state()
+
+    async def async_run_script(
+        self,
+        script: Script,
+        *,
+        run_variables: _VarsType | None = None,
+        context: Context | None = None,
+    ) -> None:
```

Comment:
Do we need to add this to the base class already in this PR; do you have a draft or something already waiting adding trigger support to other template entities?

Reply:
I've been on the fence about this.  I want to make a base class for both trigger based and template based template entities that has this function.  Something like an abstract class that manages a collection of template actions.  Then all template entities would inherit that class.  I have not made a PR for that yet because it would require me to touch all existing template entities.  Should I do that before this PR or after?

Reply:
Please do it after.